ush/forecast_postdet.sh

```diff
@@ -497,229 +469,198 @@ EOF
       logo=${COM_ATMOS_HISTORY}/${RUN}.t${cyc}z.atm.logf${FH3}.txt
       pgbo=${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.master.grb2f${FH3}
       flxo=${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.sfluxgrbf${FH3}.grib2
-      eval $NLN $atmo $atmi
-      eval $NLN $sfco $sfci
-      eval $NLN $logo $logi
-      if [ $WRITE_DOPOST = ".true." ]; then
-        eval $NLN $pgbo $pgbi
-        eval $NLN $flxo $flxi
+      eval "${NLN}" "${atmo}" "${atmi}"
+      eval "${NLN}" "${sfco}" "${sfci}"
+      eval "${NLN}" "${logo}" "${logi}"
+      if [[ ${WRITE_DOPOST} = ".true." ]]; then
+        eval "${NLN}" "${pgbo}" "${pgbi}"
+        eval "${NLN}" "${flxo}" "${flxi}"
       fi
     done
   else
-    for n in $(seq 1 $ntiles); do
-      eval $NLN nggps2d.tile${n}.nc       ${COM_ATMOS_HISTORY}/nggps2d.tile${n}.nc
-      eval $NLN nggps3d.tile${n}.nc       ${COM_ATMOS_HISTORY}/nggps3d.tile${n}.nc
-      eval $NLN grid_spec.tile${n}.nc     ${COM_ATMOS_HISTORY}/grid_spec.tile${n}.nc
-      eval $NLN atmos_static.tile${n}.nc  ${COM_ATMOS_HISTORY}/atmos_static.tile${n}.nc
-      eval $NLN atmos_4xdaily.tile${n}.nc ${COM_ATMOS_HISTORY}/atmos_4xdaily.tile${n}.nc
+    for n in $(seq 1 "${ntiles}"); do
+      eval "${NLN}" nggps2d.tile"${n}".nc       "${COM_ATMOS_HISTORY}"/nggps2d.tile"${n}".nc
+      eval "${NLN}" nggps3d.tile"${n}".nc       "${COM_ATMOS_HISTORY}"/nggps3d.tile"${n}".nc
+      eval "${NLN}" grid_spec.tile"${n}".nc     "${COM_ATMOS_HISTORY}"/grid_spec.tile"${n}".nc
+      eval "${NLN}" atmos_static.tile"${n}".nc  "${COM_ATMOS_HISTORY}"/atmos_static.tile"${n}".nc
+      eval "${NLN}" atmos_4xdaily.tile"${n}".nc "${COM_ATMOS_HISTORY}"/atmos_4xdaily.tile"${n}".nc
     done
   fi
 }
 
-FV3_GFS_nml(){
+FV3_nml(){
   # namelist output for a certain component
-  echo SUB ${FUNCNAME[0]}: Creating name lists and model configure file for FV3
+  echo SUB "${FUNCNAME[0]}": Creating name lists and model configure file for FV3
   # Call child scripts in current script directory
-  source $SCRIPTDIR/parsing_namelists_FV3.sh
+  source "${SCRIPTDIR}"/parsing_namelists_FV3.sh
   FV3_namelists
-  echo SUB ${FUNCNAME[0]}: FV3 name lists and model configure file created
-}
-
-DATM_nml(){
-  source $SCRIPTDIR/parsing_namelists_DATM.sh
-  DATM_namelists
-  echo SUB ${FUNCNAME[0]}: DATM name lists and model configure file created
+  echo SUB "${FUNCNAME[0]}": FV3 name lists and model configure file created
 }
 
-data_out_GFS() {
-  # data in take for FV3GFS
-  # Arguments: None
-  #
-  #------------------------------------------------------------------
-  # make symbolic links to write forecast files directly in memdir
+FV3_out() {
   echo "SUB ${FUNCNAME[0]}: copying output data for FV3"
-  #------------------------------------------------------------------
 
-  if [ $SEND = "YES" ]; then
```

Comment:
Since this whole function is copying to COM, perhaps we move the conditional to `exglobal_forecast.sh` and just not call the out functions in the first place if `$SENDCOM == "NO"`.

Reply:
could do.  we can address that later.